tests/functional/metrics/test_metrics.py

```diff
@@ -196,6 +196,42 @@ def test_simple_metric(
             run_dbt(["run"])
 
 
+invalid_metrics__missing_expression_yml = """
+version: 2
+metrics:
+  - name: number_of_people
+    label: "Number of people"
+    description: Total count of people
+    calculation_method: count
+    timestamp: created_at
+    time_grains: [day, week, month]
+    dimensions:
+      - favorite_color
+      - loves_dbt
+    meta:
+        my_meta: 'testing'
+"""
+
+
+class TestInvalidMetricMissingExpression:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "people_metrics.yml": invalid_metrics__missing_expression_yml,
+            "people.sql": models__people_sql,
+        }
+
+    # tests that we get a ParsingException with an invalid model ref, where
+    # the model name does not have quotes
```

Comment:
I think this is still true, no? This PR doesn't change anything with regards to `model` validation, only `expression`. I still get a parsing exception if I remove the model field from a definition and then run `dbt parse`

Reply:
@callum-mcdata  this specific test isn't testing an invalid ref missing quotes, it's testing a missing expression, right?  

You are right that we still expect the ParsingException, but the reason is different for this test.

Reply:
AH - I see what you mean 🤦 . Yes I updated the comments in the test to better reflect what the actual issue is AND fixed the metric spec provided to that test because it also contained a missing model which would have error'd out and not actually tested this behavior.